packages/zoe/src/contractFacet/vatRoot.js

```diff
@@ -8,51 +8,82 @@
 // `yarn build-zcfBundle`.
 
 import { Far } from '@endo/marshal';
+import { E } from '@endo/far';
 
 import '../../exported.js';
 import '../internal-types.js';
 
 import { makeZCFZygote } from './zcfZygote.js';
 
+const { details: X } = assert;
+
 /**
  * @param {VatPowers & { testJigSetter: TestJigSetter }} powers
- * @returns {{ executeContract: ExecuteContract}}
+ * @param {{contractBundleCap: BundleCap, zoeService: ZoeService, invitationIssuer: Issuer, privateArgs?: any}} vatParameters
+ * @param {import('@agoric/vat-data').Baggage} baggage
  */
-export function buildRootObject(powers) {
+export async function buildRootObject(powers, vatParameters, baggage) {
   // Currently, there is only one function, `executeContract` called
   // by the Zoe Service. However, when there is kernel support for
   // zygote vats (essentially freezing and then creating copies of
   // vats), `makeZCFZygote`, `zcfZygote.evaluateContract` and
   // `zcfZygote.startContract` should exposed separately.
   const { testJigSetter } = powers;
+  const { contractBundleCap } = vatParameters;
+  assert(
+    contractBundleCap,
+    X`expected vatParameters.contractBundleCap ${vatParameters}`,
+  );
+  let { zoeService, invitationIssuer } = vatParameters;
+  const firstTime = !baggage.has('DidStart');
```

Comment:
good solution to the condition, imo. still concerned about `DidStart=false` case but it's unlikely in practice and I don't have a better idea

Reply:
yeah! I think it's the "true" that suggests there's a "false" possibility. The repetition of the key sound more like the logic is by presence/absence. It might also help to make that a constant and add a little comment about how it's intended to be used.